app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,33 @@
+
+class CommentsController < ApplicationController
+
+
+  def create
+    session[:return_to] ||= request.referer
```

Comment:
I recommend either using the suggested method of returning to referer in this reading:
http://www.vikingcodeschool.com/unit-10-advanced-forms-and-active-record/redirection-class-methods-namespacing

Or at the minimum at least pulling these into a before_action.